py/picca/delta_extraction/expected_fluxes/true_continuum.py

```diff
@@ -307,76 +307,67 @@ def read_true_continuum(self, forest):
             f"{self.input_directory}/{healpix//100}/{healpix}/truth-{in_nside}-"
             f"{healpix}.fits")
         hdul = fits.open(filename_truth)
-        wmin = hdul["TRUE_CONT"].header["WMIN"]
-        wmax = hdul["TRUE_CONT"].header["WMAX"]
-        dwave = hdul["TRUE_CONT"].header["DWAVE"]
-        twave = np.arange(wmin, wmax + dwave, dwave)
+        lambda_min = hdul["TRUE_CONT"].header["WMIN"]
+        lambda_max = hdul["TRUE_CONT"].header["WMAX"]
+        delta_lambda = hdul["TRUE_CONT"].header["DWAVE"]
+        lambda_ = np.arange(lambda_min, lambda_max + delta_lambda, delta_lambda)
         true_cont = hdul["TRUE_CONT"].data
         hdul.close()
         indx = np.where(true_cont["TARGETID"]==forest.targetid)
-        true_continuum = interp1d(twave, true_cont["TRUE_CONT"][indx])
+        true_continuum = interp1d(lambda_, true_cont["TRUE_CONT"][indx])
 
         if Forest.wave_solution == "log":
             forest.continuum = true_continuum(10**forest.log_lambda)[0]
-            mean_optical_depth = np.ones(forest.log_lambda.size)
-            tau, gamma, lambda_rest_frame = 0.0023, 3.64, 1215.67
-            w = 10.**forest.log_lambda / (1. + forest.z) <= lambda_rest_frame
-            z = 10.**forest.log_lambda / lambda_rest_frame - 1.
+            forest.continuum *= self.get_mean_flux(10**forest.log_lambda)[0]
 
         elif Forest.wave_solution == "lin":
             forest.continuum = true_continuum(forest.lambda_)[0]
-            mean_optical_depth = np.ones(forest.lambda_.size)
-            tau, gamma, lambda_rest_frame = 0.0023, 3.64, 1215.67
-            w = forest.lambda_ / (1. + forest.z) <= lambda_rest_frame
-            z = forest.lambda_ / lambda_rest_frame - 1.
+            forest.continuum *= self.get_mean_flux(forest.lambda_)
```

Comment:
the option for log lambda above ends with [0], but not the one for linear lambda. Why?

Reply:
This is an error on my side. In both cases the [0] should not be there.